### PR TITLE
Add psutil dependency for cli

### PR DESCRIPTION
**Pull request by jonringer**
the ss2 cli script imports it:

```
$ ./results/python37Packages.pyroute2/bin/ss2
Traceback (most recent call last):
  File "/nix/store/7wsicyf6lf5z0fj8flxi5kkjg2pv1prf-python3.7-pyroute2-0.5.9/bin/.ss2-wrapped", line 28, in <module>
    import psutil
ModuleNotFoundError: No module named 'psutil'
```
https://github.com/svinota/pyroute2/blob/7d1a19c60aeceb3851413dcdcd547ee2f254bc11/cli/ss2#L26

### CODE REVIEW

**Comment by jonringer** [comment]
related Packaging PR https://github.com/NixOS/nixpkgs/pull/80691

File: setup.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
       author_email='peter@svinota.eu',
       url='https://github.com/svinota/pyroute2',
       license='dual license GPLv2+ and Apache v2',
+      install_reqiures=['psutil~=5.0'],
```
Comment (Mic92) [inline setup.py]:
```suggestion
      install_requires=['psutil~=5.0'],
```

**Comment by svinota** [comment]
I don't know how I missed this dependency merging ss2 :|

Give me a day or two to figure out how to avoid this dependency at all.

**Comment by Mic92** [comment]
You would first need to read `/proc/net/*` to get all the inodes and than map those back to processes.

**Comment by svinota** [comment]
@jonringer @Mic92 Would it work to make `ss2` an optional component for some time until I strip psutil code from the script? As it is here: https://github.com/svinota/pyroute2/commit/e4416a38d63e180ebdc13a3ceea2e3813b2f4901

There are too many users that will not be happy to have psutil in the requirements, so I can not just force to install it wildly together with the library.

File: setup.py (hunk 1 above)
Comment (jonringer) [reply setup.py]: there could be a breaking change on the 6.0 release, we would want an upper bound

**Comment by Mic92** [comment]
I suppose that would work too. In nixpkgs it would be an all or because we tend to avoid optional dependencies. So we would either drop the script for the time being or have psutil.

**Comment by jonringer** [comment]
So I'm not the maintainer of the pyroute2 packages, I just noticed this discrepancy when reviewing a package bump.

Thought upstream would like to know.

If you want to vendor the code like @Mic92 mentioned https://github.com/svinota/pyroute2/pull/679#issuecomment-589601548 or do another alternative is up to you.

> There are too many users that will not be happy to have psutil in the requirements, so I can not just force to install it wildly together with the library.

Then @Mic92's  recommendation would probably be best, unless you want to remove the related functionality

**Comment by cherusk** [comment]
ss2 - gold and silver!
@svinota , as discussed offline, let's make ss2 an extra for now.

**Comment by jonringer** [comment]
superseded by https://github.com/svinota/pyroute2/pull/680

**Pull request closed without merging**